File: backend/app/schemas/document.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict
from app.models.document import DocumentType


class DocumentCreate(BaseModel):
    name: str
    type: DocumentType
    content_md: str = ""


class DocumentUpdate(BaseModel):
    name: str | None = None
    content_md: str | None = None


class DocumentBrief(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    name: str
    type: DocumentType


class DocumentResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    user_id: str
    name: str
    type: DocumentType
    content_md: str
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/document.ts
export type DocumentType = "resume" | "cover_letter";

export interface Document {
  id: string;
  user_id: string;
  name: string;
  type: DocumentType;
  content_md: string;
  created_at: string;
  updated_at: string;
}

export interface DocumentBrief {
  id: string;
  name: string;
  type: DocumentType;
}
